Index: app/src/main/java/com/kryptgames/health/fitwithfriends/activity/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/kryptgames/health/fitwithfriends/activity/MainActivity.java	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ app/src/main/java/com/kryptgames/health/fitwithfriends/activity/MainActivity.java	(date 1566978083923)
@@ -13,9 +13,11 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import android.os.Bundle;
+import android.text.Editable;
 import android.text.InputFilter;
 import android.text.Spanned;
 import android.text.TextUtils;
+import android.text.TextWatcher;
 import android.util.Log;
 
 import android.view.View;
@@ -31,6 +33,9 @@
 
 import com.google.android.gms.tasks.OnFailureListener;
 import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
@@ -72,7 +77,14 @@
     Spinner weightSpinner;
 
     Uri imageUri;
-    TextView tv;
+    EditText emailAddress;
+
+
+    private FirebaseAuth mAuth;
+    FirebaseUser currentUser;
+
+
+    private String valid_email;
 
     FirebaseStorage storage;
     StorageReference storageReference;
@@ -90,8 +102,37 @@
         storage = FirebaseStorage.getInstance();
         storageReference = storage.getReference();
 
+
+        InputFilter alphaFilter[] = new InputFilter[]{ new InputFilter() {
+            @Override
+            public CharSequence filter(CharSequence src, int start, int end, Spanned dest, int dstart, int dend)
+            {
+                if (src.equals("")) { // for backspace
+                    return src;
+                }
+                if (src.toString().matches("[a-zA-Z ]+")) {
+                    return src;
+                }
+                return "";
+            }
+        }};
+
+        initilizeUI();
+
+        FirebaseApp.initializeApp(getApplicationContext());
+
+        mAuth = FirebaseAuth.getInstance();
+
+        currentUser = mAuth.getCurrentUser();
+        updateUI(currentUser);
+
+
+
+
         profilePic = findViewById(R.id.profilePic);
 
+        emailAddress = findViewById(R.id.emailAddress);
+
         edit = findViewById(R.id.edit);
 
         edit.setOnClickListener(new View.OnClickListener() {
@@ -143,41 +184,8 @@
         edName = findViewById(R.id.edName);
         edLastName = findViewById(R.id.edLastName);
 
-
-
-
-        edName.setFilters(new InputFilter[] {
-                new InputFilter() {
-                    public CharSequence filter(CharSequence src, int start,
-                                               int end, Spanned dst, int dstart, int dend) {
-                        if(src.equals("")){ // for backspace
-                            return src;
-                        }
-                        if(src.toString().matches("[a-zA-Z ]+")){
-                            return src;
-                        }
-                        return "";
-                    }
-                }
-        });
-
-
-        edLastName.setFilters(new InputFilter[] {
-                new InputFilter() {
-                    public CharSequence filter(CharSequence src, int start,
-                                               int end, Spanned dst, int dstart, int dend) {
-                        if(src.equals("")){ // for backspace
-                            return src;
-                        }
-                        if(src.toString().matches("[a-zA-Z ]+")){
-                            return src;
-                        }
-                        return "";
-                    }
-                }
-        });
-
-
+        edName.setFilters(alphaFilter);
+        edLastName.setFilters(alphaFilter);
 
 
 
@@ -220,7 +228,7 @@
         heightSpinner.setAdapter(meterAdapter);
 
 
-       /* List<String> Gender = new ArrayList<>();
+        List<String> Gender = new ArrayList<>();
         Gender.add(0, "Gender");
         Gender.add("Male");
         Gender.add("Female");
@@ -230,7 +238,7 @@
         ArrayAdapter<String> dataAdapter;
         dataAdapter = new ArrayAdapter(this, android.R.layout.simple_spinner_item, Gender);
         dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        spinnerGender.setAdapter(dataAdapter);*/
+        spinnerGender.setAdapter(dataAdapter);
 
 
 
@@ -300,6 +308,18 @@
         };
     }
 
+    private void updateUI(FirebaseUser currentUser) {
+
+        if (currentUser != null) {
+            Toast.makeText(MainActivity.this, "HOLA" + currentUser.getEmail(), Toast.LENGTH_SHORT).show();
+        } else {
+            Toast.makeText(MainActivity.this, "SORRY I DONT KNOW U YET", Toast.LENGTH_SHORT).show();
+
+        }
+
+
+    }
+
 
     private void profileCreated() {
 
@@ -341,6 +361,57 @@
         }
     }
 
+
+    private void initilizeUI() {
+        // TODO Auto-generated method stub
+
+        emailAddress = (EditText) findViewById(R.id.emailAddress);
+
+        emailAddress.addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before,
+                                      int count) {
+                // TODO Auto-generated method stub
+
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count,
+                                          int after) {
+                // TODO Auto-generated method stub
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                // TODO Auto-generated method stub
+
+                // TODO Auto-generated method stub
+                Is_Valid_Email(emailAddress); // pass your EditText Obj here.
+            }
+
+            public void Is_Valid_Email(EditText edt) {
+                if (edt.getText().toString() == null) {
+                    edt.setError("Invalid Email Address");
+                    valid_email = null;
+                } else if (isEmailValid(edt.getText().toString()) == false) {
+                    edt.setError("Invalid Email Address");
+                    valid_email = null;
+                } else {
+                    valid_email = edt.getText().toString();
+                }
+            }
+
+            boolean isEmailValid(CharSequence email) {
+                return android.util.Patterns.EMAIL_ADDRESS.matcher(email)
+                        .matches();
+            } // end of TextWatcher (email)
+        });
+
+    }
+
+
     private void uploadImage() {
 
         if(imageUri != null)
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ app/src/main/res/layout/activity_main.xml	(date 1566978084010)
@@ -122,6 +122,22 @@
 
         </LinearLayout>
 
+        <EditText
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/r_dimen"
+            android:layout_marginStart="@dimen/h_dimen"
+            android:layout_marginLeft="@dimen/l_dimen"
+            android:layout_marginTop="@dimen/h_dimen"
+            android:layout_marginEnd="@dimen/h_dimen"
+            android:background="@drawable/edittext"
+            android:hint="Email Address"
+            android:padding="@dimen/c_dimen"
+            android:textAlignment="center"
+            android:inputType="text"
+            android:textSize="@dimen/h_dimen"
+            android:id="@+id/emailAddress"/>
+
+
         <TextView
             android:id="@+id/mDisplayDate"
             android:layout_width="match_parent"
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ app/src/main/res/values/strings.xml	(date 1566799245462)
@@ -7,7 +7,6 @@
     <string name="app_name3">CreateProfile</string>
 
 
-
     <array name="Gender">
         <item>Gender</item>
         <item>Male</item>
@@ -15,9 +14,8 @@
         <item>Others</item>
     </array>
 
-
-
-
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
 
 
 </resources>
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ build.gradle	(date 1566901543538)
@@ -7,7 +7,7 @@
         
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.2'
+        classpath 'com.android.tools.build:gradle:3.5.0'
         classpath 'com.google.gms:google-services:4.2.0'
         
         // NOTE: Do not place your application dependencies here; they belong
Index: .idea/codeStyles/Project.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/codeStyles/Project.xml	(date 1566906935772)
+++ .idea/codeStyles/Project.xml	(date 1566906935772)
@@ -0,0 +1,116 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <codeStyleSettings language="XML">
+      <indentOptions>
+        <option name="CONTINUATION_INDENT_SIZE" value="4" />
+      </indentOptions>
+      <arrangement>
+        <rules>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>xmlns:android</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>^$</XML_NAMESPACE>
+                </AND>
+              </match>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>xmlns:.*</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>^$</XML_NAMESPACE>
+                </AND>
+              </match>
+              <order>BY_NAME</order>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>.*:id</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>http://schemas.android.com/apk/res/android</XML_NAMESPACE>
+                </AND>
+              </match>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>.*:name</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>http://schemas.android.com/apk/res/android</XML_NAMESPACE>
+                </AND>
+              </match>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>name</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>^$</XML_NAMESPACE>
+                </AND>
+              </match>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>style</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>^$</XML_NAMESPACE>
+                </AND>
+              </match>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>.*</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>^$</XML_NAMESPACE>
+                </AND>
+              </match>
+              <order>BY_NAME</order>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>.*</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>http://schemas.android.com/apk/res/android</XML_NAMESPACE>
+                </AND>
+              </match>
+              <order>ANDROID_ATTRIBUTE_ORDER</order>
+            </rule>
+          </section>
+          <section>
+            <rule>
+              <match>
+                <AND>
+                  <NAME>.*</NAME>
+                  <XML_ATTRIBUTE />
+                  <XML_NAMESPACE>.*</XML_NAMESPACE>
+                </AND>
+              </match>
+              <order>BY_NAME</order>
+            </rule>
+          </section>
+        </rules>
+      </arrangement>
+    </codeStyleSettings>
+  </code_scheme>
+</component>
\ No newline at end of file
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ app/src/main/AndroidManifest.xml	(date 1566974436604)
@@ -2,8 +2,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.kryptgames.health.fitwithfriends">
 
-    <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
 
     <application
         android:name=".FitWithFriendsApplication"
@@ -13,10 +13,12 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".activity.PhoneAuthenticationActivity"
-            android:theme="@style/NoActionBar"></activity>
+        <activity android:name=".activity.WelcomeActivity"></activity>
         <activity
-            android:name=".activity.WelcomeActivity"
+            android:name=".activity.PhoneAuthenticationActivity"
+            android:theme="@style/NoActionBar" />
+        <activity
+            android:name=".activity.MainActivity"
             android:theme="@style/NoActionBar">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ app/build.gradle	(date 1566901503425)
@@ -30,9 +30,12 @@
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
     implementation 'com.google.firebase:firebase-auth:16.2.1'
     implementation 'com.google.firebase:firebase-database:16.1.0'
+
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'androidx.test:runner:1.2.0'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0-alpha02'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0-alpha02'
 
     implementation 'com.google.android.material:material:1.0.0'
     implementation 'com.github.mancj:MaterialSearchBar:0.8.2'
@@ -48,14 +51,11 @@
     implementation 'com.github.mayuce:LottieAlertDialog:1.1'
 
 
-    implementation 'com.google.firebase:firebase-auth:16.0.5'
-
     implementation 'com.google.firebase:firebase-storage:16.0.4'
 
     implementation 'de.hdodenhof:circleimageview:3.0.0'
 
 
-
 }
 
 apply plugin: 'com.google.gms.google-services'
Index: gradle/wrapper/gradle-wrapper.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>windows-1252
===================================================================
--- gradle/wrapper/gradle-wrapper.properties	(revision bfd133339f04ed465ca0db780b7091fae64b7efa)
+++ gradle/wrapper/gradle-wrapper.properties	(date 1566901671246)
@@ -3,4 +3,4 @@
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
